exporter/azuremonitorexporter/trace_to_envelope.go

```diff
@@ -0,0 +1,679 @@
+func copyAttributesWithoutMapping(
+	attributeMap pdata.AttributeMap,
+	properties map[string]string,
+	measurements map[string]float64) {
+
+	copyAndMapAttributes(attributeMap, properties, measurements, nil)
+}
+
+// Attribute extraction logic for HTTP Span attributes
+func copyAndExtractHTTPAttributes(
+	attributeMap pdata.AttributeMap,
+	properties map[string]string,
+	measurements map[string]float64) *HTTPAttributes {
+
+	attrs := &HTTPAttributes{}
+	copyAndMapAttributes(
+		attributeMap,
+		properties,
+		measurements,
+		func(k string, v pdata.AttributeValue) { attrs.MapAttribute(k, v) })
+
+	return attrs
+}
+
+// Attribute extraction logic for RPC Span attributes
+func copyAndExtractRPCAttributes(
+	attributeMap pdata.AttributeMap,
+	properties map[string]string,
+	measurements map[string]float64) *RPCAttributes {
+
+	attrs := &RPCAttributes{}
+	copyAndMapAttributes(
+		attributeMap,
+		properties,
+		measurements,
+		func(k string, v pdata.AttributeValue) { attrs.MapAttribute(k, v) })
+
+	return attrs
+}
+
+// Attribute extraction logic for Database Span attributes
+func copyAndExtractDatabaseAttributes(
+	attributeMap pdata.AttributeMap,
+	properties map[string]string,
+	measurements map[string]float64) *DatabaseAttributes {
+
+	attrs := &DatabaseAttributes{}
+	copyAndMapAttributes(
+		attributeMap,
+		properties,
+		measurements,
+		func(k string, v pdata.AttributeValue) { attrs.MapAttribute(k, v) })
+
+	return attrs
+}
+
+// Attribute extraction logic for Messaging Span attributes
+func copyAndExtractMessagingAttributes(
+	attributeMap pdata.AttributeMap,
+	properties map[string]string,
+	measurements map[string]float64) *MessagingAttributes {
+
+	attrs := &MessagingAttributes{}
+	copyAndMapAttributes(
+		attributeMap,
+		properties,
+		measurements,
+		func(k string, v pdata.AttributeValue) { attrs.MapAttribute(k, v) })
+
+	return attrs
+}
+
+func idToHex(source []byte) string {
+	if source == nil {
+		return ""
+	}
+
+	return fmt.Sprintf("%02x", source)
+}
+
+func formatSpanDuration(span pdata.Span) string {
+	startTime := toTime(span.StartTime())
+	endTime := toTime(span.EndTime())
+	return formatDuration(endTime.Sub(startTime))
+}
+
+// Maps incoming Span to a type defined in the specification
+func mapIncomingSpanToType(attributeMap pdata.AttributeMap) spanType {
+	// No attributes
+	if attributeMap.Len() == 0 {
+		return unknownSpanType
+	}
+
+	// HTTP
+	if _, exists := attributeMap.Get(conventions.AttributeHTTPMethod); exists {
+		return httpSpanType
+	}
+
+	// RPC
+	if _, exists := attributeMap.Get(conventions.AttributeRPCSystem); exists {
+		return rpcSpanType
+	}
+
+	// Database
+	if _, exists := attributeMap.Get(attributeDBSystem); exists {
+		return databaseSpanType
+	}
+
+	// Messaging
+	if _, exists := attributeMap.Get(conventions.AttributeMessagingSystem); exists {
+		return messagingSpanType
+	}
+
+	// if _, exists := attributeMap.Get(conventions.AttributeFaaSTrigger); exists {
+	// 	return faasSpanType
+	// }
```

Comment:
Doesn't need commented right? I see that `faasSpanType` is handled in `spanToEnvelope`

Reply:
correct. uncommented.